test/suites/basic.sh

```diff
@@ -672,6 +672,29 @@ test_basic_usage() {
   lxc profile delete foo
   lxc delete -f c1
 
+  # Test assigning a profile through a YAML file to an instance.
+  poolName=$(lxc profile device get default root pool)
+  cat > "test_profile.yaml" << EOF
+config:
+  limits.cpu: 2
+  limits.memory: 1024MB
+description: Test profile
+devices:
+  root:
+    path: /
+    pool: ${poolName}
+    type: disk
+EOF
+
+  lxc profile create foo < test_profile.yaml
+  lxc init testimage c1
+  lxc profile assign c1 foo
+  lxc config show c1 --expanded | grep -q "limits.cpu: \"2\""
+  lxc config show c1 --expanded | grep -q "limits.memory: 1024MB"
+  lxc delete -f c1
+  lxc profile delete foo
+  rm "test_profile.yaml"
```

Comment:
Do we need the iterim file, shall we just print the profile text straight into the profile command?

Reply:
sure, let's do that. It's simpler after all.

Reply:
fixed